jsonnet/components/control-plane.libsonnet

```diff
@@ -141,6 +141,43 @@ function(params)
       },
     },
 
+    // This adds a kubelet ServiceMonitor for special use with
+    // prometheus-adapter if enabled by the configuration of the cluster monitoring operator.
+    serviceMonitorKubeletResourceMetrics: self.serviceMonitorKubelet {
+      metadata+: {
+        name: 'kubelet-resource-metrics',
+      },
+      spec+: {
+        endpoints:
+          std.filterMap(
+            function(e)
+              'path' in e && e.path == '/metrics/cadvisor'
+            ,
+            function(e)
+              e {
+                path: '/metrics/resource',
+                honorTimestamps: true,
+                metricRelabelings: [
+                  {
```

Comment:
```suggestion
                  // Keep container_cpu_usage_seconds_total and container_memory_working_set_bytes metrics, this is all that the Prometheus adapter needs. Node metrics are provided by node_exporter (Linux) or Windows exporter.
                  // The scrape_error metric can be useful for troubleshooting.
                  {
```

Reply:
:+1: Lets be friendly to our future selves.